docs/pipelines.md

```diff
@@ -272,11 +272,77 @@ spec:
 ```
 
 #### Param enum
-> :seedling: **Specifying `enum` is an [alpha](additional-configs.md#alpha-features) feature.** The `enable-param-enum` feature flag must be set to `"true"` to enable this feature.
+> :seedling: **`enum` is an [alpha](additional-configs.md#alpha-features) feature.** The `enable-param-enum` feature flag must be set to `"true"` to enable this feature.
 
-> :seedling: This feature is WIP and not yet supported/implemented. Documentation to be completed.
+Parameter declarations can include `enum` which is a predefine set of valid values that can be accepted by the `Pipeline` `Param`. For example, the valid/allowed values for `Param` "message" is bounded to `v1` and `v2`:
 
-Parameter declarations can include `enum` which is a predefine set of valid values that can be accepted by the `Pipeline`.
+``` yaml
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: pipeline-param-enum
+spec:
+  params:
+  - name: message
+    enum: ["v1", "v2"]
+    default: "v1"
```

Comment:
do we have validation that the default value is in the enum? mind pointing me to it?

it may be obvious, but please let's add docs making it clear that it's not just that values passed to the pipelinerun needs to be in the enum but also the default value

Reply:
Hi Jerop, yes we do: 
https://github.com/tektoncd/pipeline/blob/140b633d8e9ca07f81f236a5a234c8665f7058ca/pkg/apis/pipeline/v1/param_types.go#L165C1-L169C4

I will update the doc to make it explicit

Reply:
Doc updated for both pipelines and tasks, PTAL.